Formula/mariadb@10.4.rb

```diff
@@ -36,7 +36,9 @@ class MariadbAT104 < Formula
   keg_only :versioned_formula
 
   # See: https://mariadb.com/kb/en/changes-improvements-in-mariadb-104/
-  deprecate! date: "2024-06-01", because: :unsupported
+  # and https://mariadb.org/about/#maintenance-policy
+  # End-of-life on 2024-06-18. Disable date set to 3 months after.
+  deprecate! date: "2024-09-18", because: :unsupported
```

Comment:
Use official EOL date from https://mariadb.org/about/#maintenance-policy +3 months for disable date to allow users to get final release while also scheduling removal given large number of `mariadb` versioned formulae.

Release series | Stable (GA) Date | End-of-life
-- | -- | --
10.4 | 18 Jun 2019 | **18 Jun 2024**